rust/chains/hyperlane-ethereum/src/rpc_clients/mod.rs

```diff
@@ -0,0 +1,93 @@
+use ethers::providers::HttpClientError;
+use tracing::{info, trace, warn};
+
+pub use self::{fallback::*, retrying::*};
+
+mod fallback;
+mod retrying;
+
+enum CategorizedResponse<R> {
+    IsOk(R),
+    /// An error that is (probably) not our fault
+    RetryableErr(HttpClientError),
+    /// retryable error, but caller should backoff
+    RateLimitErr(HttpClientError),
+    /// An error that will (probably) keep happening no matter how many times we
+    /// retry
+    NonRetryableErr(HttpClientError),
+}
+
+const METHODS_TO_NOT_RETRY: &[&str] = &["eth_estimateGas"];
+const METHOD_TO_NOT_RETRY_WHEN_NOT_SUPPORTED: &[&str] = &["eth_feeHistory"];
```

Comment:
Another case where we shouldn't retry `eth_feeHistory`, experienced by the Khalini ppl:
```
{"timestamp":"2023-03-29T10:59:31.447872Z","level":"INFO","fields":{"message":"JsonRpcError in http provider","attempt":5,"next_backoff_ms":800,"error":"(code: -32602, message: invalid type: integer `10`, expected a (both 0x-prefixed or not) hex string with length between (0; 64] at line 1 column 2, data: None)"},"target":"hyperlane_ethereum::retrying","span":{"method":"eth_feeHistory","name":"request_with_retry"}
```